cmd/storer2elastic/dataprocessor/helpers.go

```diff
@@ -0,0 +1,239 @@
+		Marshalizer:             dp.marshalizer,
+		Hasher:                  dp.hasher,
+		Shuffler:                dataIndexerDisabled.NewNodesShuffler(),
+		EpochStartNotifier:      &disabled.EpochStartNotifier{},
+		BootStorer:              memDB,
+		ShardIDAsObserver:       shardID,
+		NbShards:                nodesConfig.NumberOfShards(),
+		EligibleNodes:           eligibleValidators,
+		WaitingNodes:            waitingValidators,
+		SelfPublicKey:           []byte("own public key"),
+		ConsensusGroupCache:     consensusGroupCache,
+		ShuffledOutHandler:      disabled.NewShuffledOutHandler(),
+	}
+	baseNodesCoordinator, err := sharding.NewIndexHashedNodesCoordinator(argsNodesCoordinator)
+	if err != nil {
+		return nil, fmt.Errorf("%w while creating nodes coordinator", err)
+	}
+
+	return baseNodesCoordinator, nil
+}
+
+func (dp *dataProcessor) preparePersistersHolder(dbInfo *databasereader.DatabaseInfo) (*persistersHolder, error) {
+	persHold := &persistersHolder{}
+
+	shardHeadersPersister, err := dp.databaseReader.LoadPersister(dbInfo, "BlockHeaders")
+	if err != nil {
+		return nil, err
+	}
+	persHold.shardHeadersPersister = shardHeadersPersister
+
+	miniBlocksPersister, err := dp.databaseReader.LoadPersister(dbInfo, "MiniBlocks")
+	if err != nil {
+		return nil, err
+	}
+	persHold.miniBlocksPersister = miniBlocksPersister
+
+	txsPersister, err := dp.databaseReader.LoadPersister(dbInfo, "Transactions")
+	if err != nil {
+		return nil, err
+	}
+	persHold.transactionPersister = txsPersister
+
+	uTxsPersister, err := dp.databaseReader.LoadPersister(dbInfo, "UnsignedTransactions")
+	if err != nil {
+		return nil, err
+	}
+	persHold.unsignedTransactionsPersister = uTxsPersister
+
+	rTxsPersister, err := dp.databaseReader.LoadPersister(dbInfo, "RewardTransactions")
+	if err != nil {
+		return nil, err
+	}
+	persHold.rewardTransactionsPersister = rTxsPersister
+
+	return persHold, nil
+}
+
+func (dp *dataProcessor) closePersisters(persisters *persistersHolder) {
+	err := persisters.shardHeadersPersister.Close()
+	log.LogIfError(err)
+
+	err = persisters.miniBlocksPersister.Close()
+	log.LogIfError(err)
+
+	err = persisters.transactionPersister.Close()
+	log.LogIfError(err)
+
+	err = persisters.unsignedTransactionsPersister.Close()
+	log.LogIfError(err)
+
+	err = persisters.rewardTransactionsPersister.Close()
+	log.LogIfError(err)
+}
+
+func getMetaChainDatabasesInfo(records []*databasereader.DatabaseInfo) ([]*databasereader.DatabaseInfo, error) {
+	metaChainDBsInfo := make([]*databasereader.DatabaseInfo, 0)
+	for _, record := range records {
+		if record.Shard == core.MetachainShardId {
+			metaChainDBsInfo = append(metaChainDBsInfo, record)
+		}
+	}
+
+	if len(metaChainDBsInfo) == 0 {
+		return nil, ErrNoMetachainDatabase
+	}
+
+	return metaChainDBsInfo, nil
+}
+
+func getShardDatabaseForEpoch(records []*databasereader.DatabaseInfo, epoch uint32, shard uint32) (*databasereader.DatabaseInfo, error) {
```

Comment:
this can be part of a some kind of collection that retain the database info instances

Reply:
this would translate also to a function call. Can remain like this for now